x-pack/platform/plugins/shared/fleet/server/services/agents/action_runner.ts

```diff
@@ -221,9 +221,11 @@ export abstract class ActionRunner {
 
     const getAgents = async () => {
       const namespaceFilter = await agentsKueryNamespaceFilter(this.actionParams.spaceId);
-      const kuery = namespaceFilter
-        ? `${namespaceFilter} AND ${this.actionParams.kuery}`
-        : this.actionParams.kuery;
+
+      const kuery = [
```

Comment:
This helps support empty string kuery values and not end up with a dangling `AND`